activerecord/lib/active_record/transactions.rb

```diff
@@ -356,6 +356,7 @@ def add_to_transaction
       if has_transactional_callbacks?
         self.class.connection.add_transaction_record(self)
       else
+        sync_with_transaction_state
```

Comment:
If we are always going to be cleaning the state before adding the record to a new transaction, why do we need to hold multiple levels on `_start_transaction_state` ?
I wonder if this is breaking something else we dont have tests for it. Something around .destroy on a multi level transaction.